lib/session-state.js

```diff
@@ -244,7 +252,8 @@ module.exports = class SessionState {
 
       this._moveToCore(src.core)
 
-      await oldStorage.close()
+      if (this.lingers === null) this.lingers = []
+      this.lingers.push(oldStorage)
```

Comment:
we could add to lingers at the start of the scope and avoid assigning oldStorage

Reply:
nice